src/sasl/cyrsasl_external.erl

```diff
@@ -124,8 +126,12 @@ get_xmpp_addresses(Cert) ->
 get_credentials(Cred, Key) ->
     mongoose_credentials:get(Cred, Key, undefined).
 
-get_username(<<"">>) ->
-    <<"">>;
-get_username(Jid) ->
+verify_server(Jid, Server) ->
     JidRecord = jid:binary_to_bare(Jid),
-    JidRecord#jid.user.
+    RequestServer = JidRecord#jid.server,
+    case RequestServer of
```

Comment:
I think `JidRecord#jid.server` could be used inside the `case ... of`. Also it's safer to use `lserver` as this field is normalized according to XMPP specs.